sos/policies/redhat.py

```diff
@@ -284,6 +295,45 @@ def check(cls):
                         return True
         return False
 
+    def prompt_for_upload_user(self):
+        if self.commons['cmdlineopts'].upload_user:
+            return
+        # Not using the default, so don't call this prompt for RHCP
+        if self.commons['cmdlineopts'].upload_url:
+            super(RHELPolicy, self).prompt_for_upload_user()
+            return
+        if self.case_id:
+            self.upload_user = input(_(
+                "Enter your Red Hat Customer Portal username (empty to use "
+                "public dropbox): ")
+            )
+
+    def get_upload_url(self):
+        if self.commons['cmdlineopts'].upload_url:
+            return self.commons['cmdlineopts'].upload_url
+        if (not self.case_id or not self.upload_user or not
+                self.upload_password):
+            # Cannot use the RHCP. Use anonymous dropbox
+            self.upload_user = self._upload_user
+            self.upload_directory = self._upload_directory
+            self.upload_password = None
+            return RH_FTP_HOST
+        else:
+            if self.case_id and self.upload_user and self.upload_password:
```

Comment:
Isnt this test redundant? (opposite to "if (not self.case_id or not ..)" in else branch, so..).

Reply:
It is, I think here I was initially planning some more logical checks, but ended up not needing them and didn't fully clean it up. Fixing.